File: raeunlee/week_17/PGS_조이스틱.py
def solution(name):
    # 알파벳 변경
    up_down = 0
    # 커서 이동, 이름의 길이 - 1
    left_right = len(name) - 1  
    
    for i, updown in enumerate(name):
    	# 알파벳 변경 횟수, 위아래 중 최소 이동
        up_down += min(ord(updown) - ord('A'), ord('Z') - ord(updown) + 1)
        
        next = i + 1
        while next < len(name) and name[next] == 'A':
            next += 1
            
        left_right = min([ left_right, 2 * i + len(name) - next, i + 2 * (len(name) - next) ])
        
        
   
    return up_down + left_right

File: OneK-2/week_17/PGS_조이스틱.java
import java.util.*;

class Solution {
    public int solution(String name) {
        int answer = 0;
        int len = name.length();
        int move = len - 1;

        for (int i = 0; i < len; i++) {
            char c = name.charAt(i);
            //상하
            answer += Math.min(c - 'A', 'Z' - c + 1);
            //좌우
            int next = i + 1;
            while (next < len && name.charAt(next) == 'A') {
                next++;
            }
            move = Math.min(move, Math.min((i * 2) + len - next, (len - next) * 2 + i));
        }
        return answer + move;
    }
}